Used open graph protocol to add site meta data

SirChronicle opened the pull request:
parameters used:

title: "Null NEU"
description: "A cybersecurity club at Northeastern University focused on learning and sharing knowledge in security."

Parthiv-M commented on layouts/partials/opengraph.html:
```diff
@@ -0,0 +1,11 @@
+{{ with .Site.Params.title }}
+<meta property="og:title" content="{{ . }}" />
+{{ end }}
```
Can be simplified to `<meta property="og:title" content="{{ .Site.Params.title }}" />`

Parthiv-M commented on layouts/partials/opengraph.html:
```diff
@@ -0,0 +1,11 @@
+{{ with .Site.Params.title }}
+<meta property="og:title" content="{{ . }}" />
+{{ end }}
+{{ with .Site.Params.description }}
+<meta property="og:description" content="{{ . }}" />
+{{ end }}
```
Same as above

Parthiv-M reviewed (changes requested):
The approach is spot on, just some more improvements!

- Would be helpful if we can add the following tags

  ```
  <meta name="description" content="{{ if .Description }}{{ .Description }}{{ else }}{{ .Site.Params.description }}{{ end }}" />
  
  <meta name="twitter:title" content="{{ .Title }} | {{ .Site.Title }}" />
  <meta name="twitter:image" content="{{ .Site.Params.og_image | absURL }}" />
  
  <meta itemprop="name" content="{{ .Title }} | {{ .Site.Title }}" />
  <meta name="application-name" content="{{ .Title }} | {{ .Site.Title }}" />
  
  <meta property="og:title" content="{{ .Title }} | {{ .Site.Title }}" />
  <meta property="og:type" content="website" />
  <meta property="og:title" content="{{ .Site.Title }}" />
  <meta property="og:description"content="{{ if .Description }}{{ .Description }}{{ else }}{{ .Site.Params.description }}{{ end }}" />
  <meta property="og:site_name" content="{{ .Site.Title }}" />
  <meta property="og:url" content="{{ .Permalink }}" />
  <meta property="og:locale" content="en">
  <meta property="og:image" content="{{ .Params.Site.BaseURL }}{{ .Site.Params.og_image | relURL }}" />
  <meta property="og:image:secure_url" content="{{ .Site.Params.og_image | absURL }}" />
  <meta property="og:type" content="website" />
  ```

This is just something I got off another resource, please validate and modify accordingly before including in our partial.

SirChronicle commented:
I've added the suggested meta tags after reviewing each (commented all tags for images for future updates) and updated the relevant files. Let me know if there's anything else to adjust!
